module/VuFind/src/VuFind/Log/Logger.php

```diff
@@ -51,110 +52,227 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Site
  */
-class Logger extends BaseLogger
+
+
+class Logger implements LoggerInterface
 {
     /**
      * Is debug logging enabled?
      *
-     * @var bool
+     * @var bool`
      */
-    protected $debugNeeded = false;
+    protected bool $debugNeeded = false;
 
     /**
      * User IP address reader
      *
      * @var UserIpReader
      */
-    protected $userIpReader;
+    protected UserIpReader $userIpReader;
+    protected MonologLogger $monologLogger;
+
+    protected const LEVEL_MAP = [
+        'err'       => LogLevel::ERROR,
+        'warn'      => LogLevel::WARNING,
+    ];
+
+    public function __construct(UserIpReader $userIpReader, ?MonologLogger $monologLogger)
```

Comment:
You could also use constructor property promotion for minor simplification here:

```suggestion
    public function __construct(protected UserIpReader $userIpReader, ?MonologLogger $monologLogger)
```